content/en/blog/falcosidekick-reponse-engine-part-7-cloud-functions.md

```diff
@@ -0,0 +1,367 @@
+--member="serviceAccount:${SA_ACCOUNT}@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com" \
+--role="roles/cloudfunctions.invoker"
+```
+
+At the beginning, we already enabled [WorkloadIdentity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) feature for our GKE Cluster by setting `--workload-pool` flag. What we need to do here is we should add a `iam.workloadIdentityUser` role for the given Service Account. 
+
+```bash
+$ gcloud iam service-accounts add-iam-policy-binding \
+  --role roles/iam.workloadIdentityUser \
+  --member "serviceAccount:${GOOGLE_PROJECT_ID}.svc.id.goog[${FALCO_NAMESPACE}/falco-falcosidekick]" \
+  ${SA_ACCOUNT}@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com
+```
+
+We need to _annotate_ the `falco-falcosidekick` resource. So it can grant access for our Cluster. Set up the Falcosidekick SA to impersonate a GCP SA:
+```bash
+$ kubectl annotate serviceaccount \
+  --namespace $FALCO_NAMESPACE \
+  falco-falcosidekick \
+  iam.gke.io/gcp-service-account=${SA_ACCOUNT}@${GOOGLE_PROJECT_ID}.iam.gserviceaccount.com
+```
+
+### Create Necessary Cluster Role
+
+To limit function role access in the particular cluster, we need to ensure our SA only have limited permissions within a particular namespace by using [Role Bindings](https://kubernetes.io/docs/reference/access-authn-authz/rbac/#rolebinding-and-clusterrolebinding). It should only access _Pod_ resource for _DELETE_ action:
+
+```bash
+$ kubectl create serviceaccount pod-destroyer
+$ kubectl create clusterrole pod-destroyer 
+  --verb=delete 
+  --resource=pod  # give only pod resource access for delete op 
+$ kubectl create clusterrolebinding pod-destroyer 
+  --clusterrole pod-destroyer 
+  --serviceaccount default:pod-destroyer
+```
+
+To obtain the Token from secret, we need to get `pod-deleter` ServiceAccount resource first:
+
+```bash
+$ POD_DESTROYER_TOKEN=$(kubectl get secrets $(kubectl get serviceaccounts pod-deleter -o json \
+  | jq -r '.secrets[0].name') -o json \
+  | jq -r '.data.token' \
+  | base64 -D)
+```
+
+Add the `pod-destroyer` user to your _KUBECONFIG_:
+
+```bash
+# Generate your KUBECONFIG
+$ kubectl config view  --minify --flatten > kubeconfig_pod-destroyer.yaml
+
+# Set the token at the end of yaml
+$ cat << EOF >> kubeconfig_pod-destroyer.yaml
+users:
+- name: user.name
+  user:
+    token: $POD_DESTROYER_TOKEN
+```
+
+We can test it with [auth can-i](https://kubernetes.io/docs/reference/access-authn-authz/authorization/#checking-api-access) to check if roles are set correctly
+```bash
+$ kubectl auth can-i list deployments  # no
+$ kubectl auth can-i delete pod  # yes
+$ kubectl access-matrix  # github.com/corneliusweig/rakkess
+```
+
+### Create Secret Manager
+
+Where Secret Manager get involved our architecture is we had to find a way out to initialize our _kubeclient_ in our function. Simply, we need to store our `pod-destroyer`'s _KUBECONFIG_ and access from the function.
```

Comment:
It is unclear what you mean by "Where Secret Manager get involved our architecture". Please rewrite for clarity.